### PR TITLE
[DOC] update changelog and git blame ignore

### LINKED ISSUE #3752: Update changelog for PRs related to PEP8 compliance
- [ ] add changelog entry to point to all the PRs that have not yet been referenced and concern PEP8 / isort compliance
- [ ] update gitblame ignore

### PR DESCRIPTION
**Pull request by Remi-Gau**
Closes #3752
Relates to #3653 

Changes proposed in this pull request:

- update git blame ignore
- update changelog with changes for pep8 and isort compliance

### CODE REVIEW

**Comment by Remi-Gau** [comment]
Related to https://github.com/nilearn/nilearn/pull/3701#discussion_r1170076459

```bash
 git blame --ignore-revs-file .git-blame-ignore-revs nilearn/image/resampling.py 
```

Still shows content from 3651 event though it is supposed to be ignored:

```
# run isort and flake8 on the whole nilearn code base in CI (#3651)
44933309ecd09add795ce1c3ea5edaab4294da87
```

More annoyingly it is also still the case for:

```bash
 git blame --ignore-rev 44933309  nilearn/image/resampling.py
```

**Comment by Remi-Gau** [comment]
Behavior is the same on the github interface that supposedly picks up on git blame ignore:

https://github.com/Remi-Gau/nilearn/blame/ignore_blame/nilearn/image/resampling.py

**Comment by Remi-Gau** [comment]
things I have tried:

- only having the commit shasum in the file with no comment (help identify invalid commit number but does not fix the issue)

File: nilearn/plotting/tests/test_matrix_plotting.py
Comment (Remi-Gau) [inline nilearn/plotting/tests/test_matrix_plotting.py:1]: Will put this fix into a separate PR

**Comment by Remi-Gau** [comment]
I still cannot make the git blame ignore work properly to ignore the PRs listed in it but at least the PRs for flake8 and isort compliance are listed

**Review by ymzayek — APPROVED** [review]
@Remi-Gau it LGTM! I think then we can leave #3653 open with a comment that we still need to get it to work. But I'm ok for merging this PR

**Comment by Remi-Gau** [comment]
yup that was my idea.

Merging this then.
